### PR TITLE
Callbacks and model_name tests

### PR DESCRIPTION
**Pull request by mfloresnss**
* Test model_name default and custom configuration in config_spec.rb
* Test model_name behaviour inside group actions in group_spec.rb
* Refactor #destroy and implement #update in group_spec.rb
* Test invoke callbacks inside #destroy and #update
* Test callbacks in a model, parameters and multiple method definition
in callbacks_spec.rb
* Fix splat operator in arguments passed to update_sync for
delayed_job_spec.rb

### CODE REVIEW

File: spec/lib/callbacks_spec.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  end
+
+  let(:documents) do
+    Array.new(3) { DocumentClass.create }
+  end
+  let(:document_type) { DocumentClass.to_s }
+  let(:document_ids) { documents.map(&:id).map(&:to_s) }
+  let(:owner_id) { 'owner_id' }
+  let(:update_params) { {signature: 'Group signature'} }
+  let(:group) { ActAsGroup::Group.create(type: document_type, ids: document_ids, owner_id: owner_id) }
```
Comment (pacop) [inline spec/lib/callbacks_spec.rb]: This line has more than 100 characters

File: spec/lib/group_spec.rb [hunk 2]
```diff
@@ -81,83 +81,217 @@
 
     let(:group) { ActAsGroup::Group.create(type: document_type, ids: document_ids, owner_id: owner_id) }
```
Comment (pacop) [inline spec/lib/group_spec.rb]: I know that line wasn't modified by you, but do you mind breaking the line to have less than 100 characters? Thanks!

File: spec/lib/group_spec.rb [hunk 3]
```diff
@@ -81,83 +81,217 @@
...
+          ActAsGroup.configure do
+            authorize? { true }
+          end
+        end
+
+        context 'when resources are checked', :run_delayed_jobs do
+          it 'all resources are sent to successful callback' do
+            expect(Post).to receive(:invoke_after_successful_group_update)
+              .with(group, update_params)
+              update_group
```
Comment (pacop) [inline spec/lib/group_spec.rb]: This line doesn't seem to be correctly idented.

File: spec/lib/group_spec.rb [hunk 4]
```diff
@@ -81,83 +81,217 @@
...
-          destroy_resource :custom_destroy
+          authorize? { |document, _method, user| document.authorized && user.present? }
         end
       end
 
-      it 'not authorized resources are sent to failed callback', :run_delayed_jobs do
-        expect(Post).not_to receive(:invoke_after_failed_group_delete)
-        group.destroy
+      let(:user) { User.create }
+      let(:group) { ActAsGroup::Group.create(type: document_type, ids: document_ids, owner_id: user.id) }
```
Comment (pacop) [inline spec/lib/group_spec.rb]: This line has more than 100 characters

File: spec/lib/group_spec.rb [hunk 5]
```diff
@@ -81,83 +81,217 @@
...
+          authorize? { |document, _method, user| document.authorized && user.is_a?(Admin) }
+          model_name 'Admin'
         end
       end
-    end
-  end
 
-  describe '#update' do
+      let(:admin) { Admin.create }
+      let(:group) { ActAsGroup::Group.create(type: document_type, ids: document_ids, owner_id: admin.id) }
```
Comment (pacop) [inline spec/lib/group_spec.rb]: This line has more than 100 characters

**Comment by mfloresnss** [comment]
@pacop done

**Pull request merged**